core/src/dpdk/port.rs

```diff
@@ -437,26 +474,57 @@ impl<'a> PortBuilder<'a> {
             }
 
             // configures the TX queue with defaults
-            let txq_index = TxQueueIndex(idx as u16);
+            let txq = TxQueueIndex(idx as u16);
             unsafe {
                 ffi::rte_eth_tx_queue_setup(
                     self.port_id.0,
-                    txq_index.0,
+                    txq.0,
                     self.txd,
                     socket_id.0 as raw::c_uint,
                     ptr::null(),
                 )
                 .to_result()?;
             }
 
-            let queue = PortQueue {
-                port_id: self.port_id,
-                rxq_index,
-                txq_index,
-                kni: kni.as_ref().map(|v| v.txq()),
-            };
+            let mut q = PortQueue::new(self.port_id, rxq, txq);
+
+            if let Some(kni) = &kni {
+                q.set_kni(kni.txq());
+            }
+
+            #[cfg(feature = "metrics")]
```

Comment:
any reason not to place this as a function call in port/stats.rs?

Reply:
port stats doesn't have the `idx`, and it gets created later on. and same goes for the counter, the port queue doesn't have the port name.

Reply:
I think I was more thinking of it as a function you pass what you need here into (this would still be the call site)

Reply:
that will get messy too with too many functions taking in arguments without clear context, when ideally you want to pass around finished structs not loose arguments to awkwardly named helper/utility calls. that's why I'm trying to apply the builder pattern as much as possible, but so far with kinda mixed results. let's sleep on this for now, and think about how this giant fn can be better broken down.